From 1315772a647293895a69cdd3d3168156e7b74218 Mon Sep 17 00:00:00 2001
From: Dmitry Timoshkov <dmitry@baikal.ru>
Date: Fri, 20 Jan 2017 17:08:19 +0800
Subject: [PATCH 258/341] winex11.drv: Don't add MWM_DECOR_BORDER to windows
 without a caption.

Also don't use MWM_DECOR_RESIZEH, window resizing is controlled by MWM_FUNC_RESIZE.

This patch fixes double caption observed under some WMs (Metacity, its
successor Mate, and XFCE), they always treat such windows as requesting
the caption decoration.

Wine-Bug: https://bugs.winehq.org/show_bug.cgi?id=42117
Signed-off-by: Dmitry Timoshkov <dmitry@baikal.ru>
---
 dlls/winex11.drv/window.c | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/dlls/winex11.drv/window.c b/dlls/winex11.drv/window.c
index 14da3e97b8b..cb741c509fe 100644
--- a/dlls/winex11.drv/window.c
+++ b/dlls/winex11.drv/window.c
@@ -510,11 +510,7 @@ static unsigned long get_mwm_decorations_for_style( DWORD style, DWORD ex_style
         if (style & WS_SYSMENU) ret |= MWM_DECOR_MENU;
         if (style & WS_MINIMIZEBOX) ret |= MWM_DECOR_MINIMIZE;
         if (style & WS_MAXIMIZEBOX) ret |= MWM_DECOR_MAXIMIZE;
-        if (style & WS_THICKFRAME) ret |= MWM_DECOR_RESIZEH;
     }
-    if (ex_style & WS_EX_DLGMODALFRAME) ret |= MWM_DECOR_BORDER;
-    else if (style & WS_THICKFRAME) return ret;
-    else if ((style & (WS_DLGFRAME|WS_BORDER)) == WS_DLGFRAME) ret |= MWM_DECOR_BORDER;
     return ret;
 }
 
-- 
2.50.1

